scenarios/typescript/node/captioning/caption.ts

```diff
@@ -0,0 +1,313 @@
+{
+    var caption = "";
+    if (!userConfig.partialResultsEnabled && userConfig.srtEnabled)
+    {
+        caption += `${sequenceNumber}${newline}`;
+    }
+    caption += `${TimestampFromSpeechRecognitionResult(result, userConfig)}${newline}`;
+    caption += `${result.text}${newline}${newline}`
+    return caption;
+}
+
+function WriteToConsole(text : string, userConfig : UserConfig) : void
+{
+    if (!userConfig.suppressOutputEnabled)
+    {
+        process.stdout.write(text);
+    }
+}
+
+function WriteToConsoleOrFile(text : string, userConfig : UserConfig) : void
+{
+    WriteToConsole(text, userConfig);
+    if (null !== userConfig.outputFile)
+    {
+        fs.appendFileSync(userConfig.outputFile, text);
+    }
+}
+
+function Initialize(userConfig : UserConfig) : void
+{
+    if (null !== userConfig.outputFile && fs.existsSync(userConfig.outputFile))
+    {
+        fs.unlinkSync(userConfig.outputFile);
+    }
+
+    if (!userConfig.srtEnabled)
+    {
+        WriteToConsoleOrFile(`WEBVTT${newline}${newline}`, userConfig);
+    }
+}
+
+/*
+Main functions
+*/
+
+// Note: We verified args.length >= 3 in the caller, main().
+function UserConfigFromArgs(args : string[]) : UserConfig
+{
+    return {
+        profanityFilterRemoveEnabled : CmdOptionExists(args, "-f"),
+        profanityFilterMaskEnabled : CmdOptionExists(args, "-m"),
+        inputFile : GetCmdOption(args, "-i"),
+        outputFile : GetCmdOption(args, "-o"),
+        phraseList : GetCmdOption(args, "-p"),
+        suppressOutputEnabled : CmdOptionExists(args, "-q"),
+        partialResultsEnabled : CmdOptionExists(args, "-u"),
+        stablePartialResultThreshold : GetCmdOption(args, "-r"),
+        srtEnabled : CmdOptionExists(args, "-s"),
+        trueTextEnabled : CmdOptionExists(args, "-t"),
+        subscriptionKey : args[args.length - 2],
+        region : args[args.length - 1],
+    };
+}
+
+function AudioConfigFromUserConfig(userConfig : UserConfig) : sdk.AudioConfig
+{
+    if (null !== userConfig.inputFile)
+    {
+        return sdk.AudioConfig.fromWavFileInput(fs.readFileSync(userConfig.inputFile));
+    }
+    else
+    {
+        return sdk.AudioConfig.fromDefaultMicrophoneInput();
+    }
+}
+
+function SpeechConfigFromUserConfig(userConfig : UserConfig) : sdk.SpeechConfig
+{
+    const speechConfig = sdk.SpeechConfig.fromSubscription(userConfig.subscriptionKey, userConfig.region);
+
+    if (userConfig.profanityFilterRemoveEnabled)
+    {
+        speechConfig.setProfanity(sdk.ProfanityOption.Removed);
+    }
+    else if (userConfig.profanityFilterMaskEnabled)
+    {
+        speechConfig.setProfanity(sdk.ProfanityOption.Masked);
+    }
+    
+    if (null !== userConfig.stablePartialResultThreshold)
+    {
+        speechConfig.setProperty("SpeechServiceResponse_StablePartialResultThreshold", userConfig.stablePartialResultThreshold);
```

Comment:
Use [PropertyId](https://github.com/microsoft/cognitive-services-speech-sdk-js/blob/master/src/sdk/PropertyId.ts#L307) instead of string.
```suggestion
        speechConfig.setProperty(sdk.PropertyId.SpeechServiceResponse_StablePartialResultThreshold, userConfig.stablePartialResultThreshold);
```